Kernel/Arch/x86/common/Processor.cpp

```diff
@@ -491,15 +491,15 @@ const DescriptorTablePointer& Processor::get_gdtr()
     return m_gdtr;
 }
 
-Vector<FlatPtr, 32> Processor::capture_stack_trace(Thread& thread, size_t max_frames)
+ErrorOr<Vector<FlatPtr, 32>> Processor::capture_stack_trace(Thread& thread, size_t max_frames)
```

Comment:
Would it not be a better approach to return the incomplete stack trace than to return none at all?

Reply:
would it? I don't see how knowing that we're in a function, but now how we got there is really helpful.

Reply:
I guess it could actually be confusing to get an incomplete backtrace without any indication of an error.